iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/TreeFactory.java

```diff
@@ -38,4 +40,8 @@ public FileTree file(Optional<List<InstructionTree>> instructions, Optional<Synt
   public FromTree from(SyntaxToken token) {
     return new FromTreeImpl();
   }
+
+  public MaintainerTree maintainer(SyntaxToken maintainerToken, SyntaxToken authorToken) {
+    return new MaintainerTreeImpl(maintainerToken, authorToken);
```

Comment:
Based on the example from the ticket the instruction allows more tokens `MAINTAINER Bob BobFullName bob@bob.com` Or is only the first token a valid token and the rest is dropped? If yes, we have to handle them as well. If not, we should also store them.

Reply:
Indeed it is taking all tokens -> [https://www.java4coding.com/contents/docker/docker-maintainer-command](https://www.java4coding.com/contents/docker/docker-maintainer-command)
Actually in the second `SyntaxToken`, you will have all the arguments including the spaces : `Bob BobFullName bob@bob.com`
This is coming from the way I defined it in the grammar : I'm matching everything else until end of line.

I choose to do it this way but I'm not really sure if that's a good idea. Should I go the other way as we will do for every other method, meaning we catch every token separately and store them like this ?
I'm also not sure how are handled multiple spaces, couldn't find any example. I guess it's also a good idea to keep them.

Reply:
As mentioned in slack, I changed this in the new commit to have each token separately, as we will probably do for other instructions.